ads1015/src/main/java/com/example/upm/androidthings/driversamples/ADCActivity.java

```diff
@@ -1,39 +1,97 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.example.upm.androidthings.driversamples;
 
 import android.app.Activity;
 import android.os.AsyncTask;
 import android.os.Bundle;
 import android.os.Environment;
 import android.util.Log;
+import android.widget.TextView;
 
 import com.example.upm.androidthings.driversupport.BoardDefaults;
 
 import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
 
 import mraa.mraa;
 
 public class ADCActivity extends Activity {
-
     private static final String TAG = "ADCActivity";
-    File filename = new File(Environment.getExternalStorageDirectory() + "/filename.txt");
 
-    static boolean running = true;
     static int id = 0; // Sample number
-
+    File filename = new File(Environment.getExternalStorageDirectory() + "/filename.txt");
     upm_ads1x15.ADS1015 adc;
 
-    FileWriter fw = null;
-    BufferedWriter bw = null;
+    FileWriter fw;
+    BufferedWriter bw;
+
+    TextView tv;
+    Runnable adcSensorTask = new Runnable() {
+
+        @Override
+        public void run() {
+            // Moves the current thread into the background
+            android.os.Process.setThreadPriority(android.os.Process.THREAD_PRIORITY_BACKGROUND);
+            float lastSample;
+            int i = 0;
+            try {
+                fw = new FileWriter(filename);
+                bw = new BufferedWriter(fw);
+            } catch (IOException e) {
+                Log.e(TAG, "Failed to open output file for writing: ", e);
+                System.exit(1);
+            }
+            try {
+                for (i = 0; i < 10; i++) {
+                    lastSample = adc.getLastSample();
+                    bw.write(id + " " + String.format("%.7f", lastSample) + "\n");
+                    updateUI(lastSample);
+                    id++;
+                    Thread.sleep(1000);
+                }
+
+            } catch (IOException e) {
+                Log.e(TAG, "Failed to open output file for writing: ", e);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } finally {
+                if (i >= 10) {
```

Comment:
This test for i>=10 ... what purpose does it serve?
How about:
if (bw) ...
if (fw) ...